lib/foreman/renderer/source/snapshot.rb

```diff
@@ -16,8 +16,8 @@ def snapshot_variants(template)
             Dir["#{File.join(SNAPSHOTS_DIRECTORY, template_path(template))}*"]
           end
 
-          def snapshot_path(template)
-            File.join(SNAPSHOTS_DIRECTORY, "#{template_path(template)}.snap.txt")
+          def snapshot_path(template, suffix = 'host')
+            File.join(SNAPSHOTS_DIRECTORY, "#{template_path(template).tr(' ', '_')}.#{suffix}.snap.txt")
           end
```

Comment:
This change removes spaces from snapshot text files converting them to underscore.